classes/author/Collector.inc.php

```diff
@@ -83,60 +126,42 @@ public function filterByCountry($country)
     /**
      * Filter by the specified affiliation code
      *
-     * @return \PKP\services\queryBuilders\PKPAuthorQueryBuilder
      * */
-    public function filterByAffiliation($affiliation)
+    public function filterByAffiliation($affiliation): self
     {
         $this->affiliation = $affiliation;
         return $this;
     }
 
     /**
-     * Set publicationIds filter
-     *
-     * @param array|int $publicationIds
-     *
-     * @return \PKP\services\queryBuilders\PKPAuthorQueryBuilder
+     * Limit the number of objects retrieved
      */
-    public function filterByPublicationIds($publicationIds)
+    public function limit(int $count): self
     {
-        $this->publicationIds = is_array($publicationIds) ? $publicationIds : [$publicationIds];
+        $this->count = $count;
         return $this;
     }
 
     /**
-     * @copydoc PKP\services\queryBuilders\interfaces\EntityQueryBuilderInterface::getCount()
+     * Offset the number of objects retrieved, for example to
+     * retrieve the second page of contents
      */
-    public function getCount()
+    public function offset(int $offset): self
     {
-        return $this
-            ->getQuery()
-            ->select('a.author_id')
-            ->get()
-            ->count();
-    }
-
-    /**
-     * @copydoc PKP\services\queryBuilders\interfaces\EntityQueryBuilderInterface::getIds()
-     */
-    public function getIds()
-    {
-        return $this
-            ->getQuery()
-            ->select('a.author_id')
-            ->pluck('a.author_id')
-            ->toArray();
+        $this->offset = $offset;
+        return $this;
     }
 
     /**
-     * @copydoc PKP\services\queryBuilders\interfaces\EntityQueryBuilderInterface::getQuery()
+     * @copydoc CollectorInterface::getQueryBuilder()
      */
-    public function getQuery()
+    public function getQueryBuilder(): Builder
     {
         $this->columns = ['*', 's.locale AS submission_locale'];
```

Comment:
The `columns` property should be removed; see https://github.com/pkp/pkp-docs/issues/812#issuecomment-900639385

Reply:
Changed the ->select statement like `$q->select(['*', 's.locale AS submission_locale']);`